user/ci-environment-trusty-beta.md

```diff
@@ -0,0 +1,243 @@
+contains a number of common language runtimes, tools, and data services.
+
+_Note: the **mega** image does not currently contain all the common
+tools and services present in our Precise images. This is something
+we'll be expanding over the next couple of months_
+
+### Version control
+
+All VM images have the following pre-installed:
+
+ * A Git 1.9.x release (official Ubuntu packages)
+ * Mercurial (official Ubuntu packages)
+ * Subversion (official Ubuntu packages)
+
+
+### Compilers & Build toolchain
+
+GCC, Clang, make, autotools, cmake, scons.
+- Essentially everything from `build-essential` and **clang** from the
+`llvm-3.4` Ubuntu package.
+
+
+### Networking tools
+
+curl, wget, OpenSSL, rsync
+
+### Docker
+
+Docker 1.8.2 is installed. We use the official [Docker apt
+repository](https://blog.docker.com/2015/07/new-apt-and-yum-repos/), so
+you can easily install another version with `apt-get` if needed.
+
+`[docker-compose](https://docs.docker.com/compose/)`
+[1.4.2](https://github.com/docker/compose/releases/tag/1.4.2) is also
+installed.
+
+See our [Using Docker in Builds](http://docs.travis-ci.com/user/docker/)
+section for more details on our Docker beta and its usage.
+
+### Runtimes
+
+#### Ruby
+
+[rvm](https://rvm.io/rvm/about) is installed and we've pre-installed the following Ruby versions
+with it.
+
+
+- jruby-9.0.0.0 [ x86_64 ]
+- ruby-1.8.7-p374 [ x86_64 ]
+- ruby-1.9.3-p551 [ x86_64 ]
+- ruby-2.0.0-p647 [ x86_64 ]
+- ruby-2.1.7 [ x86_64 ]
+- ruby-2.2.3 [ x86_64 ]
+
+#### Python
+
+We pre-install the following python versions
+
+- pypy 2.6.1
+- pypy3 2.4.0
+- python 2.6.9
+- python 2.7.10
+- python 3.2.6
+- python 3.3.6
+- python 3.4.3
+- python 3.5.0
+
+[pyenv](https://github.com/yyuu/pyenv#simple-python-version-management-pyenv) is also installed.
+
+#### Node.JS
+
+[nvm](https://github.com/creationix/nvm#installation) is installed and we've pre-installed the following Node versions
+with it
+
+- iojs-v1.6.4
+- v0.6.21
+- v0.8.28
+- v0.10.40
+- v0.11.16
+- v0.12.7
+- 4.0.0
+
+#### Go
+
+[gimme](https://github.com/travis-ci/gimme#gimme) is installed and we've pre-installed the following Go versions
+with it
+
+- go 1.0.3
+- go 1.1.2
+- go 1.2.2
+- go 1.3.3
+- go 1.4.2
+- go 1.5.1
+
+#### JVM
+
+- We install the latest versions OpenJDK versions from the official Ubuntu
+Trusty packages.
+- We install the latest Oracle JDK versions from Oracle.
+- [jdk_switcher](https://github.com/michaelklishin/jdk_switcher#what-jdk-switcher-is) is installed if you need another version.
+- gradle 2.5 is installed.
+- Maven 3.3.3 is installed.
+- leiningen 2.5.1 is installed.
+- sbt 0.13.8
+
+#### PHP
+
+We do not currently have any PHP versions pre-installed in the **mega**
+image. This is at the top of our list for our next update.
+
+You can manually install[phpenv](https://github.com/phpenv/phpenv#phpenv---php-multi-version-installation-and-management-for-humans) and then it will be available to install any PHP versions you need.
```

Comment:
I am not if this would work. `language: php` will force the build script to be compiled as PHP build script, which would run `phpenv` before any of the user-defined steps can be run.